### DISCUSSION TITLE
Is it possible to calculate Median using Map/Reduce Index?

### DISCUSSION THREAD

**Discussion by bobskigit** [ravendb/ravendb#13529] [Q&A]
Is it possible to calculate Median using Map/Reduce Index?

**Comment by arekpalinski** [comment]
Yes, you can store an collection of values as the map field and then union that field in the reduce part and calculate the median.

Note that in map-reduce indexes are processed in batches so be aware of `Common pitfalls with MapReduce indexes`:

https://ravendb.net/learn/inside-ravendb-book/reader/4.0/11-mapreduce-and-aggregations-in-ravendb#common-pitfalls-with-mapreduce-indexes

**Reply by krivulcik** [reply]
Would that be efficient and scalable? Let's say that we have a collection with ~100M documents with multiple numeric fields, of which we want to calculate medians. Storing a collection of values stores all the values of the documents, possibly multiple times (because of the recursive reduction).

**Reply by ayende** [reply]
You can usually just store the _unique_ values and their count, then compute the median from that.
Assuming you have low cardinality of values, that is sufficient in most cases.

If you have high cardinality, you can do estimation, see also here:
https://stackoverflow.com/questions/10109514/computing-median-in-map-reduce

And here:

https://stackoverflow.com/questions/6968215/find-median-of-a-large-integer-set-using-mapreduce

**Reply by krivulcik** [reply]
Thanks for the suggestions. We have a lot of unique values, so the frequency approach won't work for us. However, although a bit cumbersome, our approach works well enough (we precompute, cache, and accept that it might be slow).
